src/Microsoft.OData.Client/EntityTracker.cs

```diff
@@ -246,7 +246,7 @@ internal override void AttachLink(object source, string sourceProperty, object t
                     case MergeOption.PreserveChanges:
                         if ((EntityStates.Added == existing.State) ||
                             (EntityStates.Unchanged == existing.State) ||
-                            (EntityStates.Modified == existing.State && null != existing.Target))
+                            (EntityStates.Modified == existing.State && existing.Target != null))

```

Comment:
@xuzhg (Trivial) Switch order of enum operands of == operator

Reply:
changed. Thanks.